lib/jekyll/jekyll-sitemap.rb

```diff
@@ -8,20 +8,20 @@ class JekyllSitemap < Jekyll::Generator
     # Main plugin action, called by Jekyll-core
     def generate(site)
       @site = site
-      unless sitemap_exists?
-        write
-        @site.keep_files ||= []
-        @site.keep_files << "sitemap.xml"
-      end
+      @site.pages << sitemap_content unless sitemap_exists?
     end
 
+    private
+
     INCLUDED_EXTENSIONS = %W(
       .htm
       .html
       .xhtml
       .pdf
     ).freeze
 
+    MINIFY_REGEX = %r!(>\n|})\s+!.freeze
```

Comment:
If we see a `>` followed by a single linebreak, or a `}`, we will strip all the whitespace that follows.